src/core/mount.c

```diff
@@ -1010,7 +1010,7 @@ static void mount_enter_mounting(Mount *m) {
         p = get_mount_parameters_fragment(m);
         if (p && mount_is_bind(p)) {
                 r = mkdir_p_label(p->what, m->directory_mode);
-                if (r < 0)
+                if (r < 0 && r != -EEXIST)
```

Comment:
Mmmh mkdir_p & family already return 0 if there's an EEXIST along the way, as far as I can see. What problem are you seeing precisely, and how to reproduce it?